tests/test_examples.py

```diff
@@ -0,0 +1,48 @@
+"""
+Test the force offscreen auto gui mechanism.
+"""
+
+import os
+from pathlib import Path
+import importlib
+
+import numpy as np
+
+import pytest
+import wgpu.backends.rs  # noqa
+from testutils import can_use_wgpu_lib
+
+
+if not can_use_wgpu_lib:
+    pytest.skip("Skipping tests that need the wgpu lib", allow_module_level=True)
+
+
+examples_dir = Path(__file__).parent.parent / "examples"
+examples_to_test = []
+for example_path in examples_dir.glob("*.py"):
+    example_code = example_path.read_text()
+    if "wgpu.gui.auto" in example_code and "canvas = WgpuCanvas(" in example_code:
```

Comment:
it's a little dumb but it works. any ideas to improve this (without blowing up the effort)?

Reply:
A fixture that's only true for those tests? If necessary using a separate test run with difference arguments

Reply:
Well I mean the detection mechanism. What is a good way to determine if an example can be tested or not? Not all of them leverage the auto gui mechanism.

Reply:
I'm thinking of adding a marker comment to exclude an example and just run every example that doesn't have it.

Reply:
Added an including flag.